src/classes/AffiliationMappingsService.cls

```diff
@@ -64,6 +64,29 @@ public virtual with sharing class AffiliationMappingsService {
         return this.locateAffiliationMappingsMapper().getAffiliationMappingsModel();
     }
 
+    /***************************************************************************
+    * @description Retrieves a set of duplicate accountRecordTypeFieldValues from a list of Affiliation Mapping models
+    * @param affiliationMappingsModelList The affiliation mapping model list to
+    * retrieve ids from
+    * @return The set of duplicate accountRecordTypeFieldValues.
+    ***************************************************************************/
+    public Set<String> getDuplicateAccountRecordTypeFieldValuesFromList(
```

Comment:
This is a good way to ensure you do not loop through the same record type twice. Consider changing this method to handle record type model resolution based on the name and label maps and returning a structured data type with duplicate record type models and duplicate values in the mappings.

Reply:
Invalidated by previous suggestion